src/system_structure.jl

```diff
@@ -0,0 +1,851 @@
+- `bridle_damping::Float64=0.0`: Damping coefficient for bridle points.
+- `fix_sphere::Bool=false`: If true, constrains the point to a sphere.
+
+# Returns
+- `Point`: A new `Point` object.
+"""
+function Point(idx, pos_cad, type;
+    wing_idx=1, vel_w=zeros(KVec3), transform_idx=1,
+    mass=0.0, bridle_damping=0.0, fix_sphere=false
+)
+    Point(idx, transform_idx, wing_idx, pos_cad, zeros(KVec3), zeros(KVec3),
+        vel_w, zeros(KVec3), zeros(KVec3), type, mass, bridle_damping, fix_sphere)
+end
+
+"""
+    mutable struct Group
+
+A set of bridle lines that share the same twist angle and trailing edge angle.
+
+$(TYPEDFIELDS)
+"""
+mutable struct Group
+    const idx::Int16
+    const point_idxs::Vector{Int16}
+    const le_pos::KVec3 # point which the group rotates around under wing deformation
+    const chord::KVec3 # chord vector in body frame which the group rotates around under wing deformation
+    const y_airf::KVec3 # spanwise vector in local panel frame which the group rotates around under wing deformation
+    const type::DynamicsType
+    moment_frac::SimFloat
+    damping::SimFloat
+    twist::SimFloat
+    twist_ω::SimFloat
+    tether_force::SimFloat
+    tether_moment::SimFloat
+    aero_moment::SimFloat
+end
+
+"""
+    Group(idx, point_idxs, le_pos, chord, y_airf, type, moment_frac; damping=50.0)
+
+Constructs a `Group` object representing a collection of points on a kite body that share
+a common twist deformation.
+
+A `Group` models the local deformation of a kite wing section through twist dynamics.
+All points within a group undergo the same twist rotation about the chord vector.
+
+The governing equation is:
+```math
+\\begin{aligned}
+\\tau = \\underbrace{\\sum_{i=1}^{4} r_{b,i} \\times (\\mathbf{F}_{b,i} \\cdot \\hat{\\mathbf{z}})}_{\\text{bridles}} + \\underbrace{r_a \\times (\\mathbf{F}_a \\cdot \\hat{\\mathbf{z}})}_{\\text{aero}}
+\\end{aligned}
+```
+
+![System Overview](assets/group_slice.svg)
+
+where:
+- ``\\tau`` is the total torque about the twist axis
+- ``r_{b,i}`` is the position vector of bridle point ``i`` relative to the twist center
+- ``\\mathbf{F}_{b,i}`` is the force at bridle point ``i``
+- ``\\hat{\\mathbf{z}}`` is the unit vector along the twist axis (chord direction)
+- ``r_a`` is the position vector of the aerodynamic center relative to the twist center
+- ``\\mathbf{F}_a`` is the aerodynamic force at the group's aerodynamic center
+
+The group can have two [`DynamicsType`](@ref)s:
+- `DYNAMIC`: the group rotates according to Newton's second law: ``I\\ddot{\\theta} = \\tau``
+- `QUASI_STATIC`: the rotational acceleration is zero: ``\\tau = 0``
+
+# Arguments
+- `idx::Int16`: Unique identifier for the group.
+- `point_idxs::Vector{Int16}`: Indices of points that move together with this group's twist.
+- `le_pos::KVec3`: Leading edge position in body frame.
+- `chord::KVec3`: Chord vector in body frame.
+- `y_airf::KVec3`: Spanwise vector in local panel frame.
```

Comment:
The local panel frame is not defined at http://localhost:8000/reference_frames.html . Do not use reference frames without documenting how they are defined.